### ISSUE TITLE
Sending too many metric updates on sealed segments

### ISSUE THREAD

**Issue by glounder** [pravega/pravega#1828]
**Problem description**
We have a pravega system running a scaling test for a couple of days.  It looks like the number of metrics being reported are increasing over time.  We are currently seeing around 65,000 metric points written every second.  This is slowing down our influx database. 

What I think is happening is that we are still getting updates on metrics on segments that have been sealed or no reading is being done on the segment.  As scaling continues to go up and down, we get an increase of metrics being reported every 10 seconds.

**Problem location**
Metrics

**Suggestions for an improvement**
Only send metrics updates on segments if the count has changed.  Maybe delete the write_bytes metrics once a segment has been sealed?

**Comment by arvindkandhare** [comment]
In case of sealed segments there will not be any successful appends and hence the `segment_write_bytes` will not be incremented. Do you see unchanged metrics reported periodically too?

**Comment by glounder** [comment]
Yup, I did some tcpdump and saw some references to segment numbers that should be sealed.  The current segment number is around ~2400 and I saw some segment metrics reported for 641 (for segment_write_bytes).

![screen shot 2017-08-31 at 2 07 32 pm](https://user-images.githubusercontent.com/9335655/29935839-971bd7de-8e56-11e7-9c1a-c2aa53218638.png)

My understanding with the dropwizard metrics, it just sends whatever it has in memory as a gauge to the configured statsd endpoint.  I'm assuming if Pravega isn't actively removing those metrics, then they will continue to be sent on the interval.

Also note that the 65K points per second includes our downsampling points (writing data to lower precision), so its not all pravega traffic.
